### PR TITLE
Integration of #143, #144, #146

### PR DESCRIPTION
**Pull request by blueyed**
Since they conflict I've thought that this might be helpful, in case they all are accepted/make sense.

### CODE REVIEW

File: lib/diff-so-fancy.pl [hunk 1]
```diff
@@ -56,16 +47,17 @@
 	#########################
 	if ($line =~ /^${ansi_color_regex}diff --(git|cc) (.*?)(\s|\e|$)/) {
 		$last_file_seen = $5;
-		$last_file_seen =~ s|a/||; # Remove a/
+		$last_file_seen =~ s|\w/||; # Remove a/ (and handle diff.mnemonicPrefix).
```
Comment (scottchiefbaker) [inline lib/diff-so-fancy.pl]: I think this should be change to:

    $last_file_seen =~ s|^\w/||; # Remove leading "a/" from the found file

Just add an ^ anchor

**Comment by scottchiefbaker** [comment]
This looks good to me. I made one small suggestion. If you're ok with that change (and all the tests pass) I'm fine with merging this.

**Comment by blueyed** [comment]
Great.
Added two fixups (still green), and then rebased it - hopefully it will still be green.

**Comment by scottchiefbaker** [comment]
This looks great to me. If the tests all pass I'll merge this in. I think these are good changes and will result in a more accurate d-s-f.

**Comment by scottchiefbaker** [comment]
@stevemao, @paulirish any objections before I merge this?

**Comment by stevemao** [comment]
@scottchiefbaker I'm not very update-to-date with this repo so feel free to merge it :smile:

**Pull request merged**